plugins/utils/ccc.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "ccc";
+  originalName = "ccc.nvim";
+  defaultPackage = pkgs.vimPlugins.ccc-nvim;
+
+  maintainers = [helpers.maintainers.JanKremer];
+
+  settingsOptions = {
+    default_color = helpers.defaultNullOpts.mkStr "#000000" ''
+      The default color used when a color cannot be picked. It must be HEX format.
+    '';
+
+    highlight_mode = helpers.defaultNullOpts.mkStr "bg" ''
+      Option to highlight text foreground or background. It is used to output_line and highlighter.
+      Options: `"fg" | "bg" | "foreground" | "background"`
+    '';
+
+    lsp = helpers.defaultNullOpts.mkBool true ''
+      Whether to enable nvim-lsp support. The color information is updated in the background and the result is used by `|:CccPick|` and highlighter.
+    '';
+
+    highlighter = {
+      auto_enable = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable automatically on BufEnter.
+      '';
+
+      lsp = helpers.defaultNullOpts.mkBool true ''
+        If true, highlight using nvim-lsp. If LS with the color provider is not attached to a buffer, it falls back to highlight with pickers. See also `|ccc-option-lsp|`.
```

Comment:
- `nvim-lsp` -> `LSP` 
- `If LS` -> `If a language server`
- Description may be long enough to benefit from some line breaks?
- Does it make sense to include vimdoc links here? Or should it just be "\`:CccPick\`"